src/editor.cpp

```diff
@@ -1481,6 +1506,31 @@ void Editor::createConnectionItem(MapConnection* connection, bool hide) {
     connection_edit_items.append(connection_edit_item);
 }
 
+// Hides connected map tiles that cannot be seen from the current map (beyond BORDER_DISTANCE).
+void Editor::maskNonVisibleConnectionTiles() {
```

Comment:
can you explain this please

Reply:
Yeah, so to enable this functionality I had to expand the `graphicsView` to fill the entire frame, which caused outer parts of connected maps to be displayed (tiles extending further than `BORDER_DISTANCE` from the current map). These outer tiles were previously hidden by resizing the `graphicsView`. This resize was reseting the scroll bars and preventing the zoom from anchoring on the mouse cursor. So to keep the displayed tiles the same I draw over them with the background color of the `graphicsView`, this is the 'mask'.